service/service.go

```diff
@@ -1056,3 +1056,48 @@ func setArrayConfigEnvs(ctx context.Context) error {
 
 	return nil
 }
+
+func (s *service) filterArraysByZoneInfo(storageArrays map[string]StorageArrayConfig) ([]string, error) {
+	results := make([]string, 0)
+
+	for arrayID, arrayConfig := range storageArrays {
+		arrayLabels := arrayConfig.Labels
+		nodeLabels, err := s.k8sUtils.GetNodeLabels(s.opts.NodeFullName)
+		if err != nil {
+			log.Infof("failed to get Node Labels with error '%s'", err.Error())
+		}
+
+		keepArray := true
+		if hasZoneInfo(nodeLabels) || len(arrayLabels) != 0 {
+			for arrayLabelKey := range arrayLabels {
+				if _, ok := nodeLabels[arrayLabelKey]; !ok {
+					keepArray = false
+					break
+				}
+			}
+		}
+
+		if keepArray {
+			results = append(results, arrayID)
+		}
+	}
+
+	if len(results) == 0 {
+		log.Warn("No arrays match the node labels")
+	}
+
+	return results, nil
+}
+
+// hasZoneInfo checks if the given labels contain the zone information.
+func hasZoneInfo(labels map[string]string) bool {
+	hasZoneInfo := false
+	for labelKey := range labels {
+		if strings.EqualFold(labelKey, "topology.kubernetes.io/region") || strings.EqualFold(labelKey, "topology.kubernetes.io/zone") {
```

Comment:
This is not correct. The keys are user defined so may not necessarily be what you are checking for. I do think that the topology.kubernetes.io prefix should be standardized for topology setting in order to separate from other labels. However that needs to be reviewed and agreed upon so better to check for labels.

Reply:
Agreed. This part is no longer needed after discussion the labels naming with Trevor. Removed in newer commits